addons/test_mail/tests/test_message_track.py

```diff
@@ -203,6 +203,32 @@ def patched_message_track_post_template(*args, **kwargs):
         self.assertTrue(new_partner)
         self.assertEqual(new_partner.company_id, company1)
 
+    def test_track_invalid_selection(self):
+        # Test: Check that initial invalid selection values are allowed when tracking
+        # Create a record with an initially invalid selection value
+        invalid_value = 'I love writing tests!'
+        record = self.env['mail.test.selection.tracking'].create({
+            'name': 'Test Invalid Selection Values',})
+        self.env.cr.execute(
+                """
+                UPDATE mail_test_selection_tracking
+                   SET type = %s
+                 WHERE id = %s
+                """, [invalid_value, record.id]
+                )
+        
+        # Write a valid selection value
+        record.write({
+                'type': 'first',
+            })
+        record.flush()
+
+        tracking_value = self.env['mail.tracking.value'].search([
+            ('mail_message_id', 'in', record.message_ids.ids),
+            ('field', '=', 'type')])
+
+        self.assertEqual(tracking_value.old_value_char, '')
```

Comment:
It actually falls back on `''`

Reply:
Previously it didn't, because I think it inserted the selection value "on the fly" into registry. But when inserting it with SQL, ORM doesn't do anything at all.

Reply:
strike that, it actually works when invalidating cache